benchmarks/perf-tool/okpt/test/steps/steps.py

```diff
@@ -486,22 +493,21 @@ def recall_at_r(results, neighbor_dataset, r, k):
         r: number of top results to check if they are in the ground truth k-NN
         set.
         k: k value for the query
+        query_count: number of queries
     Returns:
         Recall at R
     """
     correct = 0.0
-    query = 0
-    while True:
+    for query in range(query_count):
         true_neighbors = neighbor_dataset.read(1)
         if true_neighbors is None:
             break
         true_neighbors_set = set(true_neighbors[0][:k])
         for j in range(r):
```

Comment:
FAR: may be change r to limit or something meaningful.

Reply:
r is a technical term in recall@r. I calculate it now as the fraction of (# of top r results returned by the query are in the ground truth k set) / r.

However, I think I may have this mixed up a little bit and I will need to refactor this to follow how faiss computes it: https://github.com/facebookresearch/faiss/blob/main/faiss/AutoTune.cpp#L60-L97. I will make a separate issue for this.